### DISCUSSION TITLE
Getting error in testing the fund me contract on foundry

### DISCUSSION THREAD

**Discussion by ghost** [Cyfrin/foundry-full-course-cu#2348] [Q&A]
while running the command "forge test --fork-url $SEPOLIA_RPC_URL" in command line again and again although I have checked manier times on chatgpt, blackbox , etc and now I am frustrated. this is the error:
` ` `
Compiler run failed:
Error (9553): Invalid type for argument in function call. Invalid implicit conversion from contract AggregatorV3Interface to contract AggregatorV3Interface requested.
  --> `[src/FundMe.sol:55:62:](url)`
   |
55 |         uint256 conversionRate = ethAmount.getConversionRate(s_priceFeed); 
   |                                                              ^^^^^^^^^^^   

Error (9582): Member "priceFeed" not found or not visible after argument-dependent lookup in address.
  --> script/DeployFundMe.s.sol:12:35:
   |
12 |         address ethUsdPriceFeed = helperConfig.activeNetworkConfig().priceFeed;
   |                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
` ` `
here is code of my FundMe.sol file:
` ` ` 
// SPDX-License-Identifier: MIT
// 1. Pragma
pragma solidity 0.8.19;

// 2. Imports
import {AggregatorV3Interface} from "@chainlink/contracts/src/v0.8/shared/interfaces/AggregatorV3Interface.sol";
import {PriceConverter} from "./PriceConverter.sol";
// 3. Interfaces, Libraries, Contracts
error FundMe__NotOwner();

/**
 * @title A sample Funding Contract
 * @author Patrick Collins
 * @notice This contract is for creating a sample funding contract
 * @dev This implements price feeds as our library
 */
contract FundMe {
    // Type Declarations
    using PriceConverter for uint256;

    // State variables
    uint256 public constant MINIMUM_USD = 5 * 10 ** 18;
    address private immutable i_owner;
    address[] private s_funders;
    mapping(address => uint256) private s_addressToAmountFunded;
    AggregatorV3Interface private s_priceFeed;

    // Events (we have none!)

    // Modifiers
    modifier onlyOwner() {
        // require(msg.sender == i_owner);
        if (msg.sender != i_owner) revert FundMe__NotOwner();
        _;
    }

    // Functions Order:
    //// constructor
    //// receive
    //// fallback
    //// external
    //// public
    //// internal
    //// private
    //// view / pure

    constructor(address priceFeed) {
        s_priceFeed = AggregatorV3Interface(priceFeed);
        i_owner = msg.sender;
    }

    /// @notice Funds our contract based on the ETH/USD price
    function fund() public payable {
        uint256 ethAmount = msg.value;
        uint256 conversionRate = ethAmount.getConversionRate(s_priceFeed);
        require(conversionRate >= MINIMUM_USD, "Insufficient funds");
        s_addressToAmountFunded[msg.sender] += msg.value;
        s_funders.push(msg.sender);
    }

    function withdraw() public onlyOwner {
        for (
            uint256 funderIndex = 0;
            funderIndex < s_funders.length;
            funderIndex++
        ) {
            address funder = s_funders[funderIndex];
            s_addressToAmountFunded[funder] = 0;
        }
        s_funders = new address[](0);
        // Use transfer instead of call to avoid reentrancy
        payable(msg.sender).transfer(address(this).balance);
    }

    function cheaperWithdraw() public onlyOwner {
        address[] memory funders = s_funders;
        for (
            uint256 funderIndex = 0;
            funderIndex < funders.length;
            funderIndex++
        ) {
            address funder = funders[funderIndex];
            s_addressToAmountFunded[funder] = 0;
        }
        s_funders = new address[](0);
        payable(msg.sender).transfer(address(this).balance);
    }

    /**
     * Getter Functions
     */

    /**
     * @notice Gets the amount that an address has funded
     *  @param fundingAddress the address of the funder
     *  @return the amount funded
     */
    function getAddressToAmountFunded(
        address fundingAddress
    ) public view returns (uint256) {
        return s_addressToAmountFunded[fundingAddress];
    }

    function getVersion() public view returns (uint256) {
        return s_priceFeed.version();
    }

    function getFunder(uint256 index) public view returns (address) {
        return s_funders[index];
    }

    function getOwner() public view returns (address) {
        return i_owner;
    }

    function getPriceFeed() public view returns (AggregatorV3Interface) {
        return s_priceFeed;
    }
}
` ` `
here is code of my PriceConverter.sol:
` ` `
// SPDX-License-Identifier: MIT
pragma solidity ^0.8.18;

import {AggregatorV3Interface} from "@chainlink/contracts/src/v0.8/interfaces/AggregatorV3Interface.sol";

library PriceConverter {
    function getPrice(
        AggregatorV3Interface priceFeed
    ) internal view returns (uint256) {
        (, int256 answer, , , ) = priceFeed.latestRoundData();
        return uint256(answer);
    }

    function getConversionRate(
        uint256 ethAmount,
        AggregatorV3Interface priceFeed
    ) internal view returns (uint256) {
        uint256 ethPrice = getPrice(priceFeed);
        uint256 ethAmountInUsd = (ethPrice * ethAmount) / 1e18;
        return ethAmountInUsd;
    }
}
` ` `
please help me ASAP I am stuck in this project for last 3 days

**Comment by EngrPips** [comment]
Hello @Moodyaadi, Please read this discussion https://github.com/Cyfrin/foundry-full-course-cu/discussions/2331 It should help you breakthrough from this issue.
